azimuth/plots/dataset_warnings.py

```diff
@@ -554,82 +512,72 @@ def create_histogram_mean_std(
-        eval_per_class_stats: Mean and std dev per class on the evaluation set.
-        train_overall_stats: Mean and std dev overall on the train.
-        eval_overall_stats: Mean and std dev overall on the evaluation set.
-        divergence_mean: Alert value for the mean.
-        divergence_std: Alert value for the std dev.
+        hist_per_label_per_split: Histogram of word count per label per split.
+        agg_per_split: Mean and std dev per split.
+        agg_per_label_per_split: Mean and std dev per label per split.
+        divergence_per_label_per_agg: Alert value for the mean and std dev per label.
         class_names: List of class names.
 
     Returns:
         Plot for all classes combined, and plot per class.
 
     """
-    # Sanitize values
-    train_per_class_count, eval_per_class_count = map(
-        lambda data: np.nan_to_num(np.array(data, dtype=float)),
-        (train_per_class_count, eval_per_class_count),
-    )
-    train_per_class_stats, eval_per_class_stats = map(
-        lambda data: np.nan_to_num(np.array(data, dtype=float)),
-        (train_per_class_stats, eval_per_class_stats),
-    )
+    hist_per_split = dict()
+    for split in hist_per_label_per_split.keys():
+        # Sanitize values
+        agg_per_label = agg_per_label_per_split[split]
+        for agg in agg_per_label.keys():
+            agg_per_label[agg] = np.nan_to_num(agg_per_label[agg])
+
+        hist_per_split[split] = hist_per_label_per_split[split].sum(axis=0)
 
-    # Traces and Annotations for 'all'
-    train_count_sum = train_per_class_count.sum(axis=0)
-    eval_count_sum = eval_per_class_count.sum(axis=0)
+    # Traces and Annotations across all labels
     fig_all = create_histogram_mean_std(
-        np.array(train_count_sum),
-        np.array(eval_count_sum),
-        threshold_mean,
-        threshold_std,
-        train_overall_stats,
-        eval_overall_stats,
-        divergence_mean=None,
-        divergence_std=None,
+        hist_per_split,
+        agg_per_split,
     )
 
     # Traces and Annotations for each label
     figs_dict = {}
-    for label_id, (train_data, test_data) in enumerate(
-        zip(train_per_class_count, eval_per_class_count)
-    ):
+    for label_id in np.arange(len(class_names)):
         figs_dict[class_names[label_id]] = create_histogram_mean_std(
-            train_data,
-            test_data,
-            threshold_mean,
-            threshold_std,
-            train_per_class_stats[label_id],
-            eval_per_class_stats[label_id],
-            divergence_mean[label_id],
-            divergence_std[label_id],
-            class_names[label_id],
+            {split: value[label_id] for split, value in hist_per_label_per_split.items()},
+            {
+                split: {agg: value[label_id] for agg, value in per_split_value.items()}
+                for split, per_split_value in agg_per_label_per_split.items()
```

Comment:
The way I read the name `agg_per_label_per_split`, I would expect to loop on it in the following fashion. I'm not sure I understand what an `agg` is. Can you help me understand?
```suggestion
                split: {label: agg[label_id] for label, agg in agg_per_label.items()}
                for split, agg_per_label in agg_per_label_per_split.items()
```